lib/datasource/index.ts

```diff
@@ -181,28 +181,37 @@ async function mergeRegistries(
   return null;
 }
 
+function massageRegistryUrls(registryUrls: string[]): string[] {
+  return registryUrls.filter(Boolean).map(trimTrailingSlash);
+}
+
 function resolveRegistryUrls(
   datasource: DatasourceApi,
-  extractedUrls: string[]
+  defaultRegistryUrls: string[],
+  registryUrls: string[]
 ): string[] {
-  const { defaultRegistryUrls = [] } = datasource;
   if (!datasource.customRegistrySupport) {
-    if (is.nonEmptyArray(extractedUrls)) {
+    if (
+      is.nonEmptyArray(registryUrls) ||
+      is.nonEmptyArray(defaultRegistryUrls)
+    ) {
       logger.warn(
-        { datasource: datasource.id, registryUrls: extractedUrls },
+        { datasource: datasource.id, registryUrls, defaultRegistryUrls },
         'Custom registries are not allowed for this datasource and will be ignored'
       );
     }
-    return defaultRegistryUrls;
+    return datasource.defaultRegistryUrls || [];
```

Comment:
```suggestion
    return datasource.defaultRegistryUrls ?? [];
```

for strict null checks later